src/configLoader.js

```diff
@@ -13,35 +14,36 @@ export function configLoader(configPath) {
         return process.exit(1);
     }
 
+    let rawConfig;
     if(/(\.yml)|(\.yaml)/.test(configPath)) {
-        return yaml.safeLoad(fs.readFileSync(configPath));
+        rawConfig = yaml.safeLoad(fs.readFileSync(configPath, 'utf8'));
+    } else if (/(\.json)/.test(configPath)) {
+        rawConfig = JSON.parse(fs.readFileSync(configPath, 'utf8'));
+    } else {
+        log.error(`${configPath} is not a supported Kongfig config format. Supported formats are YAML and JSON.`);
+        process.exit(1);
     }
 
-    if (/(\.json)/.test(configPath)) {
-        return JSON.parse(fs.readFileSync(configPath));
-    }
-
-    if (/(\.js)/.test(configPath)) {
-        try {
-            let config = require(resolvePath(configPath));
-
-            if (config === null || typeof config !== 'object' || Object.keys(config).length == 0) {
-                log.error('Config file must export an object!\n' + CONFIG_SYNTAX_HELP);
+    return mapDeep(rawConfig, configValue => {
+        if (typeof configValue !== 'string') {
+            return configValue;
+        }
 
-                return process.exit(1);
+        return configValue.replace(/\$\{(.+)\}/g, (match, variableName) => {
```

Comment:
Right now this will capture up to the last `}` in the string, which seems undesirable. That could either be addressed by making `+` non-greedy:
```suggestion
        return configValue.replace(/\$\{(.+?)\}/g, (match, variableName) => {
```

or by explicitly matching all characters except the `}`:
```suggestion
        return configValue.replace(/\$\{([^}]+)\}/g, (match, variableName) => {
```

Relatedly, I think this module could use a few tests – seems like there are a few tricky cases!

Reply:
tests done and this problem should be removed!